packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts

```diff
@@ -0,0 +1,179 @@
+  const expressApp = express();
+  expressApp.use(express.json());
+  expressApp.use(express.urlencoded({ extended: false }));
+  const server = http.createServer(expressApp);
+  let addressInfo: AddressInfo,
+    address: string,
+    port: number,
+    apiHost: string,
+    plugin: PluginLedgerConnectorPolkadot,
+    keychainEntryKey: string,
+    keychainEntryValue: string,
+    keychainPlugin: PluginKeychainMemory,
+    apiClient: PolkadotApi,
+    apiConfig: Configuration,
+    contractAddress: string;
+
+  const proofSize = 131072;
+  const refTime = 6219235328;
+  const gasLimit = {
+    refTime,
+    proofSize,
+  };
+  beforeAll(async () => {
+    const pruning = pruneDockerAllIfGithubAction({ logLevel });
+    await expect(pruning).toResolve();
+  });
+  afterAll(async () => {
+    await ledger.stop();
+    await plugin.shutdownConnectionToSubstrate();
+  });
+  afterAll(async () => {
+    const pruning = pruneDockerAllIfGithubAction({ logLevel });
+    await expect(pruning).resolves.toBeTruthy();
+  });
+  afterAll(async () => await Servers.shutdown(server));
+  beforeAll(async () => {
+    await ledger.start();
+    expect(ledger).toBeTruthy();
+    keychainEntryKey = uuidv4();
+    keychainEntryValue = "//Bob";
+    keychainPlugin = new PluginKeychainMemory({
+      instanceId: uuidv4(),
+      keychainId: uuidv4(),
+      // pre-provision keychain with mock backend holding the private key of the
+      // test account that we'll reference while sending requests with the
+      // signing credential pointing to this keychain entry.
+      backend: new Map([[keychainEntryKey, keychainEntryValue]]),
+      logLevel,
+    });
+    const connectorOptions: IPluginLedgerConnectorPolkadotOptions = {
+      logLevel: logLevel,
+      pluginRegistry: new PluginRegistry({ plugins: [keychainPlugin] }),
+      wsProviderUrl: DEFAULT_WSPROVIDER,
+      instanceId: instanceId,
+    };
+    plugin = new PluginLedgerConnectorPolkadot(connectorOptions);
+    await plugin.createAPI();
+
+    const listenOptions: IListenOptions = {
+      hostname: "0.0.0.0",
+      port: 0,
+      server,
+    };
+    addressInfo = await Servers.listen(listenOptions);
+    ({ address, port } = addressInfo);
+    apiHost = `http://${address}:${port}`;
+    apiConfig = new Configuration({ basePath: apiHost });
+    apiClient = new PolkadotApi(apiConfig);
+    await plugin.registerWebServices(expressApp);
+    await plugin.getOrCreateWebServices();
+  });
+  beforeAll(async () => {
+    const rawWasm = fs.readFileSync(
+      "packages/cactus-plugin-ledger-connector-polkadot/src/test/rust/fixtures/ink/flipper.wasm",
+    );
+    const result = await apiClient.deployContractInk({
+      wasm: rawWasm.toString("base64"),
+      metadata: JSON.stringify(metadata),
+      gasLimit: gasLimit,
+      storageDepositLimit: null,
+      salt: new Uint8Array().toString(),
+      web3SigningCredential: { type: "MNEMONIC_STRING", mnemonic: "//Alice" },
```

Comment:
@AnmolBansalDEV Please make sure to define constants like this in the openapi.json file which will then also make it available for import via the generated typescript code and then you can use those code constants instead of magic strings.

Reply:
updated
https://github.com/hyperledger/cacti/blob/87ec994040fade0e7a4a690883c971baa9eb9702/packages/cactus-plugin-ledger-connector-polkadot/src/test/typescript/integration/invoke-ink-contract.test.ts#L117-L122